scene/gui/rich_text_label.cpp

```diff
@@ -1265,7 +1267,9 @@ int RichTextLabel::_draw_line(ItemFrame *p_frame, int p_line, const Vector2 &p_o
 					Ref<RichTextEffect> custom_effect = item_custom->custom_effect;
 
 					if (!custom_effect.is_null()) {
+						charfx->global_elapsed_time = global_elapsed_time;
 						charfx->elapsed_time = item_custom->elapsed_time;
+						charfx->visible_time = visible_times[l.char_offset + i];
```

Comment:
This is wrong, `l.char_offset` is a character index and `i` is a glyph index, use `glyphs[i].start` instead of `i`.

Reply:
Implementing this seems to cause an issue where lines following the first/second don't get an accurate (or, usable, for that matter) value for `visible_time`. It does make rtl text work, though. Can you test on your end to see if you can make sense of why? I still don't know the codebase well enough to determine on my own.